### PR TITLE
LPD-26912 Expiration and Review date are not saved in WC

### PR DESCRIPTION
**Pull request by DiegoHu97**
Expiration and Review date are not saved in WC

# What is this trying to solve?

[Link to ticket](https://liferay.atlassian.net/browse/LPD-26912)

Save the expiration and review date when saving Web Content with FF LPD-15596 enabled

# How am I fixing it?

Set the default expiration/review date when here is no expiration/review date instead of taking into account the displayDate

# How can you verify that it works?

Follow the steps in the ticket

**CC: @jkappler**

### CODE REVIEW

**Comment by DiegoHu97** [comment]
Hey @jkappler , i just added the tests :smile:

**Comment by DiegoHu97** [comment]
ci:test:sf

**Review by jkappler — APPROVED** [review]

**Comment by jkappler** [comment]
Change in MVCActionCommand looks good @DiegoHu97

File: modules/test/playwright/tests/journal-web/journalArticle.spec.ts [hunk 1]
```diff
@@ -970,16 +970,24 @@ scheduleTest(
...
 		await journalEditArticlePage.scheduleArticle(
 			articleTitle,
-			scheduleDate
+			publishDate,
+			undefined,
+			expirationDate,
+			reviewDate
 		);
 
 		await journalEditArticlePage.assertScheduleDate(
```
Comment (ambrinchaudhary) [inline modules/test/playwright/tests/journal-web/journalArticle.spec.ts:985]: Could we rename this to `assertScheduledArticleDates` please? It makes more sense to me.

File: modules/test/playwright/tests/journal-web/journalArticle.spec.ts [hunk 2]
```diff
@@ -970,16 +970,24 @@ scheduleTest(
 		await journalEditArticlePage.goto({siteUrl: site.friendlyUrlPath});
 
 		const articleTitle = getRandomString();
-		const scheduleDate = '9987-11-26 13:00';
+		const expirationDate = '01/01/9999';
+		const publishDate = '9987-11-26 13:00';
+		const reviewDate = '01/01/9999';
```
Comment (ambrinchaudhary) [inline modules/test/playwright/tests/journal-web/journalArticle.spec.ts:975]: It is just curious why this `expirationDate` and `reviewDate` have different format than `publishDate`... ;-)

**Review by ambrinchaudhary — CHANGES_REQUESTED** [review]
Just a minor change request @DiegoHu97, the rest LGTM.

File: modules/test/playwright/tests/journal-web/journalArticle.spec.ts (hunk 2 above)
Comment (DiegoHu97) [reply modules/test/playwright/tests/journal-web/journalArticle.spec.ts:975]: I don't know why exactly, but the publish date and the expiration/review dates pickers are different (see screenshots)

**Comment by DiegoHu97** [comment]
@ambrinchaudhary I just updated the pr, thanks for the review!

**Review by ambrinchaudhary — APPROVED** [review]

**Comment by ambrinchaudhary** [comment]
ci:test:playwright

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
@ambrinchaudhary, @DiegoHu97, the playwright issues don't seems related to the changes. Do you think we can move it forward?

**Comment by ambrinchaudhary** [comment]
> @ambrinchaudhary, @DiegoHu97, the playwright issues don't seems related to the changes. Do you think we can move it forward?

Hey @jkappler, let me check the last ci:test:relevant I just run to see if that errors are not related. Then I will forward the pull ;-)

**Comment by ambrinchaudhary** [comment]
ci:report:11433244

**Comment by ambrinchaudhary** [comment]
Hey Diego, there are some errors in [journal-web/journalArticle.spec.ts](https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/900799/testReport/journal-web_journalArticle.spec/ts) that looks suspicious to me, would you mind checking them please? 

1. [Create a web content scheduled](https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/900799//testReport/journal-web_journalArticle.spec/ts/Create_a_web_content_scheduled). This test was added in this pull 
2. [LPD-19627: Translate all fields of a Web Content based on a custom structure with repeatable fields](https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/900799//testReport/journal-web_journalArticle.spec/ts/LPD_19627__Translate_all_fields_of_a_Web_Content_based_on_a_custom_structure_with_repeatable_fields)
3. [LPD-26854: LockIndicator should have an errorState](https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/900799//testReport/journal-web_journalArticle.spec/ts/LPD_26854__LockIndicator_should_have_an_errorState) and [LPD-26856: Web content should be saved as darft after changing the title and the content](https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/900799//testReport/journal-web_journalArticle.spec/ts/LPD_26856__Web_content_should_be_saved_as_darft_after_changing_the_title_and_the_content) complains about the FF

**Comment by DiegoHu97** [comment]
Hey @ambrinchaudhary I just updtaed the pr. I've fixed the failure retaled to my pr, about the second one, will be fixed in this bug (https://liferay.atlassian.net/browse/LPD-27062). About the third one failing because of the FF, I've just removed it since we don't need it for those tests

**Comment by DiegoHu97** [comment]
ci:test:sf

**Comment by DiegoHu97** [comment]
ci:test:relevant

**Review by ambrinchaudhary — APPROVED** [review]

**Comment by ambrinchaudhary** [comment]
ci:forward:force

**Pull request closed without merging**